src/toolchain.rs

```diff
@@ -423,6 +423,22 @@ impl<'a> Toolchain<'a> {
         };
         let mut cmd = Command::new(path);
         self.set_env(&mut cmd);
+
+        // If we're running cargo and the `CARGO` environment variable is set
+        // to a rustup proxy then change `CARGO` to be the real cargo binary,
+        // but only if we know the absolute path to cargo.
+        // This works around an issue with old versions of cargo not updating
+        // the environment variable itself.
+        if Path::new(&binary).file_stem() == Some("cargo".as_ref()) && path.is_absolute() {
+            if let Some(cargo) = self.cfg.process.var_os("CARGO") {
+                if fs::read_link(&cargo)
+                    .map(|p| p.file_stem() == Some("rustup".as_ref()))
```

Comment:
I believe there's a new `.is_some_and()` which can replace this `.map().unwrap_or(false)`.

Reply:
This should really not be a separate commit.

Reply:
@djc Squashed. Returning to the merge queue.